lib/rucio/rse/rsemanager.py

```diff
@@ -414,13 +414,17 @@ def upload(rse_settings, lfns, source_dir=None, force_pfn=None, force_scheme=Non
 
                 valid = None
                 try:  # Get metadata of file to verify if upload was successful
-                    stats = protocol.stat('%s.rucio.upload' % pfn)
-                    if ('adler32' in stats) and ('adler32' in lfn):
-                        valid = stats['adler32'] == lfn['adler32']
-                    if (valid is None) and ('filesize' in stats) and ('filesize' in lfn):
-                        valid = stats['filesize'] == lfn['filesize']
-                except NotImplementedError:
-                    valid = True if rse_settings['verify_checksum'] is False else False
+                    try:
```

Comment:
The only other way I could think of doing this is an `isinstance` in the except clause, these nested try's seemed cleaner.